nixos/modules/services/monitoring/prometheus/surfboard-exporter.nix

```diff
@@ -0,0 +1,67 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.surfboardExporter;
+in {
+  options = {
+    services.prometheus.surfboardExporter = {
+      enable = mkEnableOption "prometheus surfboard exporter";
+
+      port = mkOption {
+        type = types.int;
+        default = 9239;
+        description = ''
+          Port to listen on.
+        '';
+      };
+
+      modemAddress = mkOption {
+        type = types.str;
+        default = "192.168.100.1";
+        description = ''
+          The hostname or IP of the cable modem.
+        '';
+      };
+
+      extraFlags = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = ''
+          Extra commandline options when launching the surfboard exporter.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Open port in firewall for incoming connections.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    networking.firewall.allowedTCPPorts = optional cfg.openFirewall cfg.port;
+
+    systemd.services.prometheus-surfboard-exporter = {
+      description = "Prometheus exporter for surfboard cable modem";
+      unitConfig.Documentation = "https://github.com/ipstatic/surfboard_exporter";
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        User = "nobody";
+        Restart = "always";
+        PrivateTmp = true;
+        WorkingDirectory = /tmp;
```

Comment:
Does this application creates temporary files securely? There might be other applications racing for `/tmp`. Also a dedicated user different from `nobody` would be helpful. Unfortunately we do not have the latest systemd release yet.
Otherwise runtime directories could be create automatically for `DynamicUser`.